### ISSUE TITLE
PYTHONPATH needs to be exported

### ISSUE THREAD

**Issue by sherpya** [google/enjarify#2]
the .sh script does not export PYTHONPATH, you need to export it using `export PYTHONPATH` or prepending it to exec call

**Comment by Storyyeller** [comment]
That's weird. It works when I try it. Does this depends on the system?

**Comment by dwayneyuen** [comment]
Mine also needs to export PYTHONPATH, otherwise I get this error: 
"Error while finding spec for 'enjarify.main' (<class 'ImportError'>: No module named 'enjarify')"

I'm using Python 3.4.3 on OSX Yosemite.

**Comment by Storyyeller** [comment]
Maybe it's an issue with Macs? Unfortunately, I don't have any to test on.

**Comment by sherpya** [comment]
no, it's bash (I'm on linux), you need to explicitly export or put on the exec command line, you may already have set `PYTHONPATH`, so your system does not expose the problem

**Comment by pwelyn** [comment]
I'm using Python 3.4.3 on OSX 10.10.3
ln -s "$PWD/enjarify.sh" ~/bin/enjarify
Using python3 as Python interperter
/usr/local/bin/python3: Error while finding spec for 'enjarify.main' (<class 'ImportError'>: No module named 'enjarify')

**Comment by Storyyeller** [comment]
It looks like I have PYTHONPATH set to /usr/local/buildtools/current/sitecustomize, but I can't imagine how that would make a difference.

It's weird that on your system you have to explicitly export it, while on mine, merely setting it seems to be enough.

**Comment by dwayneyuen** [comment]
Are you running from the enjarify directory or somewhere else and using a symlink?

**Comment by Storyyeller** [comment]
I'm running from elsewhere using a symlink. 

Anyway, I went ahead and added export. Does it work now?

**Comment by dwayneyuen** [comment]
Working for me. :)

**Comment by moneytoo** [comment]
Well, the output of `readlink` is blank so it prevents me from calling enjarify from any outside dir (it's same for both macOS/Linux). 

```
echo $(dirname "$(readlink "${BASH_SOURCE[0]}")")
```

prints `.`

**Comment by tiann** [comment]
readlink doesn't work for me, too. my OS version is Mac OSX 10.11; I change it to this:

> $(cd `dirname $0`; pwd)

And it worked for me.

**Comment by ybf326** [comment]
Copy the following to enjarify.sh, you can solve this problem. good luck.
================================

```shell
function guess {
	if [ -z "$PYTHON" ]; then
		result=$($1 -c "print(range)" 2>/dev/null)
		if [ "$result" = "<class 'range'>" ]; then
			PYTHON=$1
		fi
	fi
}

guess "pypy3"
guess "python3"
guess "pypy"
guess "python"

#Gets the root path of the current script
getCurrentShellRootPath()
{
oldPath=`pwd`
SOURCE="$1"
while [ -h "$SOURCE"  ]; do # resolve $SOURCE until the file is no longer a symlink
    DIR="$( cd -P "$( dirname "$SOURCE"  )" && pwd  )"
    SOURCE="$(readlink "$SOURCE")"
    [[ $SOURCE != /*  ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
done
DIR2="$( cd -P "$( dirname "$SOURCE"  )" && pwd  )"
echo ${DIR2}
cd $oldPath
}

if [ -z "$PYTHON" ]; then
	echo "Unable to find python3 on path"
else
	echo "Using $PYTHON as Python interpreter"

	# Find location of this bash script, and set its directory as the PYTHONPATH
	#export PYTHONPATH=$(dirname "$(readlink "${BASH_SOURCE[0]}")")
	#Modify the path to PYTHONPATH
	shellRootPath=`getCurrentShellRootPath $0`
	export PYTHONPATH=${shellRootPath}
	

	# Now execute the actual program
	exec $PYTHON -O -m enjarify.main "$@"
fi
```

**Comment by AzenHuang** [comment]
> # Copy the following to enjarify.sh, you can solve this problem. good luck.
> ```shell
> function guess {
> 	if [ -z "$PYTHON" ]; then
> 		result=$($1 -c "print(range)" 2>/dev/null)
> 		if [ "$result" = "<class 'range'>" ]; then
> 			PYTHON=$1
> 		fi
> 	fi
> }
> 
> guess "pypy3"
> guess "python3"
> guess "pypy"
> guess "python"
> 
> #Gets the root path of the current script
> getCurrentShellRootPath()
> {
> oldPath=`pwd`
> SOURCE="$1"
> while [ -h "$SOURCE"  ]; do # resolve $SOURCE until the file is no longer a symlink
>     DIR="$( cd -P "$( dirname "$SOURCE"  )" && pwd  )"
>     SOURCE="$(readlink "$SOURCE")"
>     [[ $SOURCE != /*  ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
> done
> DIR2="$( cd -P "$( dirname "$SOURCE"  )" && pwd  )"
> echo ${DIR2}
> cd $oldPath
> }
> 
> if [ -z "$PYTHON" ]; then
> 	echo "Unable to find python3 on path"
> else
> 	echo "Using $PYTHON as Python interpreter"
> 
> 	# Find location of this bash script, and set its directory as the PYTHONPATH
> 	#export PYTHONPATH=$(dirname "$(readlink "${BASH_SOURCE[0]}")")
> 	#Modify the path to PYTHONPATH
> 	shellRootPath=`getCurrentShellRootPath $0`
> 	export PYTHONPATH=${shellRootPath}
> 	
> 
> 	# Now execute the actual program
> 	exec $PYTHON -O -m enjarify.main "$@"
> fi
> ```

it worked! Thanks!

**Comment by biranchi2018** [comment]
Try this,
import os
os.environ['PYTHONPATH'] = "/usr/bin/python3"

**Comment by FranklinYu** [comment]
`readlink` only works if you create the symbolic link, as mentioned in README. There is no need to write your own fancy script. If you installed it your own way, just remove the `readlink` command (making `PYTHONPATH` simply `dirname "${BASH_SOURCE[0]}"`).

I have tried on macOS Mojave and the script works as expected.
